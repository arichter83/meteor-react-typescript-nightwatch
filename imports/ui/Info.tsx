import * as React from 'react';
import { withTracker } from 'meteor/react-meteor-data';
import { Links, ILink } from '../api/links';

interface IProps {
  links: ILink[]
}

class Info extends React.Component<IProps> {
  render() {
    const links = this.props.links.map(
      link => this.makeLink(link)
    );

    return (
      <div>
        <h2>Learn Meteor!</h2>
        <ul>{ links }</ul>
      </div>
    );
  }

  makeLink(link: ILink) {
    return (
      <li key={link._id} id={link._id}>
        <a href={link.url} target="_blank">{link.title}</a>
      </li>
    );
  }
}

export default withTracker(() => {
  return {
    links: Links.find().fetch(),
  };
})(Info);
