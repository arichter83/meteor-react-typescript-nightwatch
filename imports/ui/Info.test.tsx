import * as React from 'react'
import { Info } from './Info'
import * as renderer from 'react-test-renderer'

const props = {
  links: [
    {
      _id: "theid",
      url: "http://test.com",
      title: "Link",
      createdAt: new Date()
    }
  ]
}

test('<Info /> rendering links', () => {
  const component = renderer.create(<Info {...props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
