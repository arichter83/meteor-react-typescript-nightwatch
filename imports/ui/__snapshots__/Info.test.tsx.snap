// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> rendering links 1`] = `
<div>
  <h2>
    Learn Meteor!
  </h2>
  <ul>
    <li
      id="theid"
    >
      <a
        href="http://test.com"
        target="_blank"
      >
        Link
      </a>
    </li>
  </ul>
</div>
`;
